src/Input/Input.jsx

```diff
@@ -46,11 +51,28 @@ export const Input = forwardRef(
     const [inputType, setInputType] = useState(type)
 
     const inputVariant = useMemo(() => {
-      if (disabled) return 'disabled'
-      else if (error) return 'error'
+      if (error) return 'error'
+      else if (warning) return 'warning'
       else if (valid) return 'valid'
+      else if (disabled) return 'disabled'
       return variant
-    }, [disabled, error, valid, variant])
+    }, [disabled, error, warning, valid, variant])
```

Comment:
The `disabled` must take precedence.

Reply:
I don't think so. There are variants for disabled error and warning inputs. By using disabled at the end I make sure that the field has the primary state (error, warning or valid) and the disabled style is applied by the `:disabled` selector.

Reply:
If the field is disabled, it cannot be an error/warning message because you cannot act on that input to fix the error/warning.